### ISSUE TITLE
mvn compile fails on Ubuntu - tried many jdks

### ISSUE THREAD

**Issue by dubnemo** [argouml-tigris-org/argouml#69]
me:~/argouml$ mvn compile

[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] argouml-core                                                       [pom]
[INFO] argouml-core-model                                                 [jar]
[INFO] argouml-core-model-mdr                                             [jar]
[INFO] argouml-core-model-euml                                            [jar]
[INFO] argouml-app                                                        [jar]
[INFO] argouml-core-notation                                              [jar]
[INFO] argouml-core-transformer                                           [jar]
[INFO] argouml-core-umlpropertypanels                                     [jar]
[INFO] argouml-core-diagrams-activity2                                    [jar]
[INFO] argouml-core-diagrams-class2                                       [jar]
[INFO] argouml-core-diagrams-sequence2                                    [jar]
[INFO] argouml-core-diagrams-state2                                       [jar]
[INFO] argouml-core-diagrams-structure2                                   [jar]
[INFO] argouml-core-diagrams-deployment2                                  [jar]
[INFO] argouml-build                                                      [jar]
[INFO] 
[INFO] ----------------------< org.argouml:argouml-core >----------------------
[INFO] Building argouml-core 0.35.2-SNAPSHOT                             [1/15]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:1.1:enforce (enforce-maven) @ argouml-core ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (default) @ argouml-core ---
[INFO] argLine set to -javaagent:/home/dubnemo/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/home/dubnemo/argouml/target/jacoco.exec
[INFO] 
[INFO] ---------------------< org.argouml:argouml-model >----------------------
[INFO] Building argouml-core-model 0.35.2-SNAPSHOT                       [2/15]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-enforcer-plugin:1.1:enforce (enforce-maven) @ argouml-model ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (default) @ argouml-model ---
[INFO] argLine set to -javaagent:/home/dubnemo/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/home/dubnemo/argouml/src/argouml-core-model/target/jacoco.exec
[INFO] 
[INFO] --- maven-resources-plugin:2.4:resources (default-resources) @ argouml-model ---
[INFO] Using 'ISO-8859-1' encoding to copy filtered resources.
[INFO] Copying 86 resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.3.1:compile (default-compile) @ argouml-model ---
[INFO] Compiling 85 source files to /home/dubnemo/argouml/src/argouml-core-model/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Failure executing javac, but could not parse the error:
warning: [options] bootstrap class path not set in conjunction with -source 7
error: Source option 7 is no longer supported. Use 8 or later.
error: Target option 7 is no longer supported. Use 8 or later.

[INFO] 1error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for argouml-core 0.35.2-SNAPSHOT:
[INFO] 
[INFO] argouml-core ....................................... SUCCESS [  1.090 s]
[INFO] argouml-core-model ................................. FAILURE [  0.481 s]
[INFO] argouml-core-model-mdr ............................. SKIPPED
[INFO] argouml-core-model-euml ............................ SKIPPED
[INFO] argouml-app ........................................ SKIPPED
[INFO] argouml-core-notation .............................. SKIPPED
[INFO] argouml-core-transformer ........................... SKIPPED
[INFO] argouml-core-umlpropertypanels ..................... SKIPPED
[INFO] argouml-core-diagrams-activity2 .................... SKIPPED
[INFO] argouml-core-diagrams-class2 ....................... SKIPPED
[INFO] argouml-core-diagrams-sequence2 .................... SKIPPED
[INFO] argouml-core-diagrams-state2 ....................... SKIPPED
[INFO] argouml-core-diagrams-structure2 ................... SKIPPED
[INFO] argouml-core-diagrams-deployment2 .................. SKIPPED
[INFO] argouml-build ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.885 s
[INFO] Finished at: 2024-02-07T14:42:14-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.1:compile (default-compile) on project argouml-model: Compilation failure
[ERROR] Failure executing javac, but could not parse the error:
[ERROR] warning: [options] bootstrap class path not set in conjunction with -source 7
[ERROR] error: Source option 7 is no longer supported. Use 8 or later.
[ERROR] error: Target option 7 is no longer supported. Use 8 or later.
[ERROR] 
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :argouml-model

**Comment by dubnemo** [comment]
Changed to default-java:
JAVA_HOME=/usr/lib/jvm/default-java

**Comment by dubnemo** [comment]
mvn test fails miserably.

Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 61

repeat....
[testOutput.txt](https://github.com/argouml-tigris-org/argouml/files/14201876/testOutput.txt)


See attached.

**Comment by argouml-automatic-verifier** [comment]
Yes, this is very much depending on the java version. The setting of compilesource is 1.7 in parentpom. I have it working, with all the tests working in ibmjava-8, and openjdk-8, 11, and 17 (amazoncorretto). We should move to compilesource 8 to match ibmjava-8. I will compile and run all tests with ibmjava-8, openjdk-8, 11, 17, and 21 to verify that it is working. Should I run all tests with openjdk-19 also? Let's see how it goes with openjdk-21 first.

**Comment by linustolke** [comment]
Yes, this is very much depending on the java version. The setting of compilesource is 1.7 in parentpom. I have it working, with all the tests working in ibmjava-8, and openjdk-8, 11, and 17 (amazoncorretto). We should move to compilesource 8 to match ibmjava-8. I will compile and run all tests with ibmjava-8, openjdk-8, 11, 17, and 21 to verify that it is working. Should I run all tests with openjdk-19 also? Let's see how it goes with openjdk-21 first.

(Sorry, I wrote the comment with the wrong user first. I now re-created it with the right user).
